CISLibrary/drivers/HBridge/DRV8837.cpp

```diff
@@ -23,17 +31,60 @@ DRV8837::~DRV8837() {}
  * @param blocking will wait until operation is complete if true
  * @return mbed_error_status_t
  */
-mbed_error_status_t DRV8837::set(double value, bool blocking) {
-  return MBED_ERROR_UNSUPPORTED;
+mbed_error_status_t DRV8837::set(double value) {
+  if (value < -1.0 || value > 1.0) {
+    return MBED_ERROR_CODE_INVALID_ARGUMENT;
+  }
+
+  this->wake();
+
+  if (value == 0) {
+    return this->stop(false);
+  } else {
+    if (value < 0) {
+      this->rev = -value;
+      this->fwd = 0.0;
+    } else {
+      this->fwd = value;
+      this->rev = 0.0;
+    }
+  }
+
+  wait_us(DRV8837_SET_WAIT_TIME_US);
+  this->sleep_immediate();
+
+  return MBED_SUCCESS;
 }
 
 /**
  * @brief Stop the output, brake or coast
  *
- * @param brake will short output if true, high impedance if true
+ * @param brake will short output if true, high impedance if false
  * @param blocking will wait until operation is complete if true
  * @return mbed_error_status_t
  */
-mbed_error_status_t DRV8837::stop(bool brake, bool blocking) {
-  return MBED_ERROR_UNSUPPORTED;
+mbed_error_status_t DRV8837::stop(bool brake) {
+  this->wake();
+  this->rev = this->fwd = (brake ? 1.0 : 0.0);
+
+  wait_us(DRV8837_SET_WAIT_TIME_US);
+  this->sleep_immediate();
+
+  return MBED_SUCCESS;
+}
+
+/**
+ * @brief Wakes the Hbridge.
+ * 
+ */
+void DRV8837::wake() {
```

Comment:
nSleep = active low sleep = 1 to wake. Read 7.3.1. Change this to enable and pass in a bool to enable or not